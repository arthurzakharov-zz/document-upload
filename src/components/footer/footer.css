.footer {
  padding-left: 20px;
  padding-right: 20px;
  display: flex;
  flex-direction: column;
  align-items: center;
  font-size: 12px;
  line-height: 22px;
  font-weight: 400;
  color: var(--grey-3);
}

.footer__line {
  margin-top: 28px;
  margin-bottom: 6px;
  border: none;
  width: 100%;
  height: 1px;
  background-color: var(--grey-4);
}

.footer__company {
  margin-bottom: 6px;
}

.footer__list {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.footer__separator {
  margin: 0 6px;
  width: 1px;
  height: 14px;
  background-color: var(--grey-3);
}

.footer__button {
  font-size: inherit;
  line-height: inherit;
  font-weight: inherit;
}

@media screen and (min-width: 768px) {
  .footer {
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: space-between;
    align-items: center;
  }

  .footer__line {
    margin-top: 35px;
    margin-bottom: 15px;
  }

  .footer__company {
    margin-bottom: 0;
  }

  .footer__list {
    flex-direction: row;
  }
}
