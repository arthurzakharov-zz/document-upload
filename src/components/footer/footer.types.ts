import type { ModalIdType } from "../../redux/modal/modal.types";

export type FooterLinkType = {
  label: string;
  modalId: ModalIdType;
};
