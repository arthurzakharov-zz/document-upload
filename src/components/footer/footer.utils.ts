export const companyName = (name: string): string => {
  const currentYear = new Date().getFullYear();
  return `Â© ${currentYear} ${name}`;
};
