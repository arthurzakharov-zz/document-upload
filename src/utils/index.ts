import { DataSizeType } from "../types";

export const isObjectEmpty = (obj: Object): boolean => {
  // eslint-disable-next-line no-restricted-syntax
  for (const prop in obj) {
    if (Object.prototype.hasOwnProperty.call(obj, prop)) {
      return false;
    }
  }
  return JSON.stringify(obj) === JSON.stringify({});
};

export const get = (obj: any, ...path: string[]): any => {
  return obj && path.reduce((result, p) => (result == null ? undefined : result[p]), obj);
};

export const convertDataSize = (number: number, from: DataSizeType = "B", to: DataSizeType = "MB"): number => {
  if (from === to) return number;
  const sizes: DataSizeType[] = ["B", "kB", "MB"];
  const fromIndex = sizes.indexOf(from);
  const toIndex = sizes.indexOf(to);
  const diff = fromIndex - toIndex;
  return diff < 0
    ? Math.round((number / 1024 ** Math.abs(diff)) * 100) / 100
    : Math.round(number * 1024 ** Math.abs(diff) * 100) / 100;
};

export const getUrlParam = (paramKey: string): string => {
  const urlSearchParams = new URLSearchParams(document.location.search);
  const paramKeyValues = urlSearchParams.getAll(paramKey);
  return paramKeyValues[paramKeyValues.length - 1] || "";
};
