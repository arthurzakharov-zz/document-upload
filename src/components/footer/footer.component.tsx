import { Fragment, useEffect, useState } from "react";
import { useDispatch } from "react-redux";
import { openModal } from "../../store/modal/modal.actions";
import { companyName } from "./footer.utils";
import { FooterLink, ModalId } from "../../types";
import "./footer.css";

function Footer() {
  const [links, setLinks] = useState<FooterLink[]>([]);

  const dispatch = useDispatch();

  useEffect(() => {
    // TODO: later will be replaced with some Redux or Context call
    setLinks([
      { label: "Allgemeine Informationen", id: "general_info" },
      { label: "Datenschutz", id: "privacy" },
      { label: "Impressum", id: "impressum" },
    ]);
  }, []);

  const linkClick = (name: ModalId): void => {
    dispatch(openModal(name));
  };

  return (
    <div className="footer">
      <hr className="footer__line" />
      <span className="footer__company">{companyName("schuldenanalyse-kostenlos.de")}</span>
      <ul className="footer__list">
        {links.map((link: FooterLink, index: number) => (
          <Fragment key={link.id}>
            <li className="footer__item">
              <button type="button" className="footer__button" onClick={() => linkClick(link.id)}>
                {link.label}
              </button>
            </li>
            {index < links.length - 1 && (
              <li className="footer__item">
                <div className="footer__separator" />
              </li>
            )}
          </Fragment>
        ))}
      </ul>
    </div>
  );
}

export default Footer;
