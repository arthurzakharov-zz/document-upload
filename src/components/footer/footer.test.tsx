import { render } from "@testing-library/react";
import Footer from "./footer.component";
import { companyName } from "./footer.utils";

describe("Footer", () => {
  test("companyName", () => {
    const year = new Date().getFullYear();
    expect(companyName("x")).toEqual(`Â© ${year} x`);
  });
  test("Screenshot", () => {
    const { container } = render(<Footer />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
