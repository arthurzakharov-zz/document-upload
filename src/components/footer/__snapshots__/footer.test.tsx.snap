// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Screenshot 1`] = `
<div
  class="footer"
>
  <hr
    class="footer__line"
  />
  <span
    class="footer__company"
  >
    Â© 2022 schuldenanalyse-kostenlos.de
  </span>
  <ul
    class="footer__list"
  >
    <li>
      <button
        class="footer__button"
        type="button"
      >
        Allgemeine Informationen
      </button>
    </li>
    <li>
      <div
        class="footer__separator"
      />
    </li>
    <li>
      <button
        class="footer__button"
        type="button"
      >
        Datenschutz
      </button>
    </li>
    <li>
      <div
        class="footer__separator"
      />
    </li>
    <li>
      <button
        class="footer__button"
        type="button"
      >
        Impressum
      </button>
    </li>
  </ul>
</div>
`;
